src/app/profiles/profiles.component.ts

```diff
@@ -0,0 +1,134 @@
+import { timeout } from 'rxjs/operators';
+import { SecurityContext, Component, Input, Output, EventEmitter, NgZone, ViewChild, AfterViewInit } from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+import { Router } from '@angular/router';
+import { ProgressService } from '../http/progress.service';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { BrowserModule } from '@angular/platform-browser';
+import { CommonModule } from '@angular/common';
+import { FormsModule } from '@angular/forms';
+import {
+  MatCardModule,
+  MatCheckboxModule,
+  MatDialogModule,
+  MatInputModule,
+  MatListModule,
+  MatPaginatorModule,
+  MatProgressBarModule,
+  MatProgressSpinnerModule,
+  MatSelectModule,
+  MatSnackBarModule,
+  MatTableModule,
+  MatTabsModule,
+  MatChipsModule,
+  MatDialog,
+  MatPaginator,
+  MatSnackBar,
+} from '@angular/material';
+
+import {MatFormFieldModule} from '@angular/material/form-field'; 
+import {AliasesLister} from '../aliases/aliases.lister';
+import {ProfilesLister} from './profiles.lister';
+import {ProfilesEditorModal} from './profiles.editor.modal';
+import {AliasesEditorModal} from '../aliases/aliases.editor.modal';
+import { RMM } from '../rmm';
+
+@Component({
+  moduleId: 'angular2/app/profiles/',
+  selector: 'profiles',
+  templateUrl: 'profiles.component.html'
+})
+
+export class ProfilesComponent implements AfterViewInit {
+  panelOpenState = false;
+  @ViewChild(MatPaginator, { static: false }) paginator: MatPaginator;
+  @Output() onClose: EventEmitter<string> = new EventEmitter();
+  domain;
+  aliases = [];
+  aliases_counter = {};
+  aliases_unique = [];
+  dialog_ref : any;
+
+  ngAfterViewInit() {
+  }
+
+  ev_reload_emiter (ev) {
+    this.load_aliases();
+    this.load_profiles();
+  }
+
+  constructor(
+    public snackBar: MatSnackBar,
+    public dialog: MatDialog,
+    public rmm: RMM,
+  ) {
+    this.rmm.runbox_domain.load();
+    this.load_profiles();
+    this.load_aliases();
+  }
+
+  load_aliases () {
+    this.rmm.alias.load();
+  }
+
+  load_profiles () {
+    this.rmm.profile.load();
+  }
+
+  show_error (message, action) {
+    this.snackBar.open(message, action, {
+      duration: 2000,
+    });
+  };
+
+  add_alias (): void {
+      window.location.href = '/mail/account_alias';
+      return;
+      let item = {};
+
+      this.dialog_ref = this.dialog.open(AliasesEditorModal, {
+          width: '600px',
+          data: item
+      });
+      this.dialog_ref.componentInstance.is_create = true;
+
+      this.dialog_ref.afterClosed().subscribe(result => {
+          this.load_aliases();
+          item = result;
+      });
+  }
+
+  add_profile (type): void {
+    let item = {type: type};
+    this.dialog_ref = this.dialog.open(ProfilesEditorModal, {
+        width: '600px',
+        data: item,
+    });
+    this.dialog_ref.componentInstance.aliases_unique = this.aliases_unique;
```

Comment:
Like the editor mode, I think this should be included in `data` above.

Reply:
Not really.

The idea here is that 'data' is the object received from the backend.

This object will populate the form which is inside the modal.

When users change the form, its reflected in 'data' which can be used to post back to the server.

The modal can have its own properties also. But they are not in data... they are properties for the modal. And the angular modal allows this.dialog_ref.componentInstance[$attr] to update them from outside (the component creating the modal in this case).